docs: revamp the README (#8052)
dagster-io/dagster: README.md

# [Dagster](https://dagster.io/) &middot; [![GitHub license](https://img.shields.io/badge/License-Apache_2.0-blue.svg)](https://github.com/dagster-io/dagster/blob/master/LICENSE) [![PyPI Version](https://badge.fury.io/py/dagster.svg)](https://pypi.org/project/dagster/) [![Coveralls coverage](https://coveralls.io/repos/github/dagster-io/dagster/badge.svg?branch=master)](https://coveralls.io/github/dagster-io/dagster?branch=master)

Dagster is an orchestration platform for the development, production, and observation of data assets.

> - **Develop and Test Locally, then Deploy Anywhere:** With Dagster, the same computations can run

Reviewer: ```suggestion
- **Develop and test locally, then deploy anywhere:** With Dagster, the same computations can run
```

Let's keep to sentence case here for consistency.

> in-process against your local file system, or on a distributed work queue against your production

Reviewer: ```suggestion
in-process against your local file system or on a distributed work queue against your production
```

Removing unnecessary comma.

> data lake. You can choose to locally develop on your labtop, deploy on-premise, or run in any

Reviewer: ```suggestion
data lake. Choose to locally develop on your laptop, deploy on-premise, or run in any
```

I vote for using an understood subject for conciseness. It also makes this sentence sound more compelling.

cloud.

> - **Model the Data Produced and Consumed:** In your orchestration graph, Dagster models data

Reviewer: ```suggestion
- **Model the data produced and consumed:** In your orchestration graph, Dagster models data
```

dependencies and handles how data passes between steps. Gradual typing on inputs and outputs catches
bugs early.

> - **Link Data To Computations:** Dagster’s Asset Catalog tracks the data sets and ML models produced

Reviewer: ```suggestion
- **Link data to computations:** Dagster’s Asset Catalog tracks the data sets and ML models produced
```

> by your jobs. You can understand how they were generated and trace issues when asset declarations

Reviewer: ```suggestion
Understand how they were generated and trace issues when asset declarations
```

Understood subject here, too.

do not match their materializations in storage.

> - **Build a Self-Service Data Platform:** Dagster helps platform teams build systems for data

Reviewer: ```suggestion
- **Build a self-service data platform:** Dagster helps platform teams build systems for data
```

practitioners. Jobs are built from shared, reusable, configurable data processing components.
Dagit, Dagster’s web interface, lets anyone inspect these objects and discover how to use them.

> - **Declare and Isolate Dependencies:** Dagster’s server model lets you isolate codebases. Problems

Reviewer: ```suggestion
- **Declare and isolate dependencies:** Dagster’s server model enables you to isolate codebases. Problems
```

in one job will not bring down the system or other jobs. Each job can have its own package
dependencies and Python version.

> - **Debug Jobs from a Rich Interface**: Dagit also includes expansive facilities for understanding

Reviewer: ```suggestion
- **Debug jobs from a rich interface**: Dagit includes expansive facilities for understanding
```

the jobs it orchestrates. When inspecting a run of your job, you can query over logs, discover the

> most time consuming tasks via a Gantt chart, re-execute subsets of steps, and more.

Reviewer: ```suggestion
most time-consuming tasks via a Gantt chart, re-execute subsets of steps, and more.
```

## Installation

> Dagster is available on PyPI, and officially supports Python 3.6+.

Reviewer: ```suggestion
Dagster is available on PyPI and officially supports Python 3.6+.
```

This installs two modules:

- **Dagster**: The core programming model.
- **Dagit**: The web interface for developing and operating Dagster jobs. It includes a DAG browser,
a type-aware interface to launch runs, a live view for in-progress runs, a catalog to view your data
assets, and more.

> For a quick overview, check out our [Getting Started](https://docs.dagster.io/getting-started) page.

Reviewer 2: I think we should have an "Example" or "Quick Start" section here to show code snippets as opposed to redirecting folks to docs site - cc @sryza / @erinkcochran87 do you have thoughts on what code to include as the starter code? - we can default to the same thing as the [Getting Started](https://docs.dagster.io/getting-started#writing-a-job) for now and switch to assets content later.

Reviewer: If we went the route of it actually being a quick start, I could be on board with it. I feel like including examples would require additional explanation that's typically too lengthy for a README.

I think it'd also be cool to add something that's similar to what [Jekyll does here](https://github.com/jekyll/jekyll#getting-started), specifically the **Getting started** and **Diving in** sections. After someone does the quick start, I think pointing them in the direction of next steps would be helpful. (FWIW, I also think we should do this in the docs - working on that 😉  )

Reviewer 2: I like the idea of having **Getting started** and **Diving in** sections. Look forward to the changes in the docs.

> ### Documentation

Reviewer 2: ```suggestion
## Documentation
```

You can find the Dagster documentation [on the website](https://docs.dagster.io).

We've divided up the documentation into several sections:

- [🌱 Tutorial](https://docs.dagster.io/tutorial/)
- [💡 Concepts](https://docs.dagster.io/concepts)
- [🚢 Deployment](https://docs.dagster.io/deployment)
- [🤝 Integrations](https://docs.dagster.io/integrations)
- [📖 Guides](https://docs.dagster.io/guides)

> ### Community

Reviewer 2: ```suggestion
## Community
```
